### ISSUE TITLE
Optimize atomic operations in scan kernel template

### ISSUE THREAD

**Issue by adamfidel** [uxlfoundation/oneDPL#1528]
It may be possible to reduce the number of atomics in the scan kernel template algorithm. Currently, we use atomic loads/stores for both the status flags and status values, but it might be possible to only use atomics for the flags and rely on the standard happens-before relations to guarantee that the values would be updated.

Discussed in https://github.com/oneapi-src/oneDPL/pull/1320#pullrequestreview-2019240324.

**Comment by danhoeflinger** [comment]
I agree that the values here shouldn't need to be atomics, and the atomics for the status flags can provide everything we need.

**Comment by danhoeflinger** [comment]
I believe this was addressed by #2357
